media-sound/mp3gain/mp3gain-1.5.2.recipe

```diff
@@ -0,0 +1,46 @@
+SUMMARY="Lossless mp3 normalizer with statistical analysis"
+DESCRIPTION="MP3Gain analyzes and adjusts mp3 files so that they have the same \
+volume. MP3Gain does not just do peak normalization, as many normalizers do. \
+Instead, it does some statistical analysis (using the replaygain algorithm) \
+to determine how loud the file actually sounds to the human ear.
+MP3Gain can adjust the volume in a completely lossless way by modifying the \
+mp3 file directly, without decoding and re-encoding. This works with all \
+mp3 players, i.e. no support for a special tag is required."
+HOMEPAGE="http://mp3gain.sourceforge.net"
+COPYRIGHT="2001-2009 David Robinson and Glen Sawyer
+	2002 John Zitterkopf"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="http://downloads.sourceforge.net/mp3gain/mp3gain-${portVersion//./_}-src.zip"
+CHECKSUM_SHA256="dbadc7a41a8a3b87d3a21a1989701cfe72d2090c3ead02b0b396a19a8acf6fca"
+SOURCE_FILENAME="mp3gain-${portVersion//./_}-src.zip"
+SOURCE_DIR=""
+
+ARCHITECTURES="?x86_gcc2 x86_64"
+
+PROVIDES="
+	mp3gain = $portVersion
+	cmd:mp3gain = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc
+	cmd:make
+	"
+
```

Comment:
```diff
 	cmd:make
 	"
+
+defineDebugInfoPackage mp3gain \
+	$binDir/mp3gain
+
 BUILD()
```
(It's nice to have the debuginfo package.)

Reply:
Well, stripping the binary is enough, who wants a debug package for every command line tool?

Reply:
I thought it was a general recommendation to create debuginfo packages whenever that was possible.
But I'm OK with not creating the debuginfo package.

That said, I think we should not decide, ourselves, which recipes should have a debuginfo package, and which should not. We can't guess what would be the needs of all the users. If someone would like to debug that program, then we should not strip it. So having the debuginfo package is the only way to satisfy all possible users.

Reply:
The paragraph about the debuginfo package in the wiki, [HaikuPorter-Guidelines#debuginfo-package](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#debuginfo-package), should probably be updated because it currently says:
> All packages should define a `debuginfo` package. This way when downloading a library user downloads only necessary data. This can be done with `defineDebugInfoPackage` macro.

The way it is written does not suggest that the macro should only be added to recipes which provide libraires. It clearly states that **all** packages should use that macro. The word "library" appears only to show that debuginfo packages are convenient.

Reply:
@korli It's nice when you could have somewhat meaningful backtrace when something gone wrong :). Besides, we've already spent time building binaries with debuginfo (`-O2 -g` is pretty common as default CFLAGS whenever the user don't have one specified, and also lengthen the time it takes to compile).